internal/core/data/v2/application/event.go

```diff
@@ -77,21 +85,21 @@ func putEventOnQueue(evt dtos.Event, ctx context.Context, dic *di.Container) {
 		ctx = context.WithValue(ctx, clients.ContentType, clients.ContentTypeJSON)
 	}
 
-	data, err = json.Marshal(evt)
+	data, err = json.Marshal(addEventReq)
 	if err != nil {
-		lc.Error(fmt.Sprintf("error marshaling V2 Event DTO: %+v", evt), clients.CorrelationHeader, correlationId)
+		lc.Error(fmt.Sprintf("error marshaling V2 AddEventRequest DTO: %+v", addEventReq), clients.CorrelationHeader, correlationId)
 		return
 	}
 
+	publishTopic := fmt.Sprintf("%s/%s/%s", configuration.MessageQueue.PublishTopicPrefix, profileName, deviceName)
 	msgEnvelope := msgTypes.NewMessageEnvelope(data, ctx)
-	err = msgClient.Publish(msgEnvelope, configuration.MessageQueue.Topic)
+	err = msgClient.Publish(msgEnvelope, publishTopic)
 	if err != nil {
-		lc.Error(fmt.Sprintf("Unable to send message for V2 API event. Correlation-id: %s, Device Name: %s, Error: %v",
-			correlationId, evt.DeviceName, err))
+		lc.Error(fmt.Sprintf("Unable to send message for V2 API event. Correlation-id: %s, Profile Name: %s, "+
+			"Device Name: %s, Error: %v", correlationId, profileName, deviceName, err))
 	} else {
 		lc.Debug(fmt.Sprintf(
-			"Event Published on message queue. Topic: %s, Correlation-id: %s ",
-			configuration.MessageQueue.Topic, correlationId))
+			"Event Published on message queue. Topic: %s, Correlation-id: %s ", publishTopic, correlationId))
```

Comment:
Change to `V2 API Event Published on message queue...`

Reply:
Updated.